Survival-DEV/blog-backend
# Blog-22: create blog meta

### :dart:  What is been done in this PR:
**endpoint** : `api/blog-meta`
- [x] Create blog_meta entity
- [x] Create CRUD ops, interface & DTO models
- [x] make the join between blogs table and blog_meta

**Relates to #22**

----
### Output Response:
at `api/blog-meta`
 > 
 ```json
  "blog_meta": {
        "id": 1,
        "key": "1234567",
        "content": "this is meta data for this blog"
    }
 ```
- for `api/blogs/:id`
```json
{
    "id": 1,
    "title": "how to be a fraud in 5 min",
    "meta_title": "in depth description about the mind of stupids",
    "slug": "/how-to-be-fraud-in-five-min",
    "summary": "something not important",
    "created_at": "2021-06-30T20:30:42.000Z",
    "updated_at": "2021-06-30T20:30:42.000Z",
    "claps": 1,
    "header_image": "wertyuiol;kjlhgkfjhdgsf",
    "published_at": "2021-06-30T20:30:42.000Z",
    "content": "ertyuiljhgkfjhdgsfdcvvbnbmnggkgk",
    "is_draft": true,
    "next_blog_id": null,
    "author_id": null,
    "blog_meta": {
        "id": 1,
        "key": "1234567",
        "content": "this is meta data for this blog"
    }
}
```
 -----

### :compass:  How to test it up
Steps to test my PR:
1. `npm ci`
2. make sure to setup your `DATABASE_URL`& `PORT` in your `.env` file
6. `nest start`
4. the active routes are-with their CRUD ops- :
   - `api/blog-meta` & `api/blog-meta/:id`
